pvlib/iotools/tmy.py

```diff
@@ -150,24 +150,30 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
 
     References
     ----------
-
     .. [1] Wilcox, S and Marion, W. "Users Manual for TMY3 Data Sets".
        NREL/TP-581-43156, Revised May 2008.
-
     .. [2] Wilcox, S. (2007). National Solar Radiation Database 1991 2005
        Update: Users Manual. 472 pp.; NREL Report No. TP-581-41364.
-    '''
-
+    .. [3] `SolarAnywhere file formats
+       <https://www.solaranywhere.com/support/historical-data/file-formats/>`_
+    """  # noqa: E501
     head = ['USAF', 'Name', 'State', 'TZ', 'latitude', 'longitude', 'altitude']
-
-    with open(str(filename), 'r') as csvdata:
-        # read in file metadata, advance buffer to second line
-        firstline = csvdata.readline()
+    try:
+        fbuf = open(str(filename), 'r')
+        firstline = fbuf.readline()  # first line contains the metadata
+    # SolarAnywhere files contain non-UTF8 characters and require
+    # encoding='iso-8859-1' on Linux in order to be parsed
+    except UnicodeDecodeError:
+        fbuf = open(str(filename), 'r', encoding='iso-8859-1')
+        firstline = fbuf.readline()  # first line contains the metadata
+    finally:
         # use pandas to read the csv file buffer
         # header is actually the second line, but tell pandas to look for
         # header information on the 1st line (0 indexing) because we've already
         # advanced past the true first line with the readline call above.
-        data = pd.read_csv(csvdata, header=0)
+        data = pd.read_csv(fbuf, header=0)
+        # close file
+        fbuf.close()
```

Comment:
I'm probably just being picky at this point, but if the `pd.read_csv` call fails for some reason then there's still risk of not closing the file.  How about this?  Better to use `with open(...):` than keep track of handle state manually. 

```python
def _parse(fbuf):
    firstline = fbuf.readline()
    data = pd.read_csv(fbuf, header=0)
    return firstline, data

try:
    with open(str(filename), 'r') as fbuf:
        firstline, data = _parse(fbuf)
except UnicodeDecodeError:
    with open(str(filename), 'r', encoding='iso-8859-1') as fbuf:
        firstline, data = _parse(fbuf)
```

Reply:
Consider it done.